### ISSUE TITLE
[BUG] Uniqueness validation fails on update

### ISSUE THREAD

**Issue by ramiromd** [phalcon/cphalcon#11828]
Hi, i have Phalcon 2.0.8 installed and i am having troubles to validate a model before update it.

The model has an uniqueness validation constraint:

```
$this->validate(new Uniqueness([
    'field' => 'slug',
    'message' => json_encode(['code' => 13, 'reason' => 'The rate plan slug must be unique.'])
]));
```

So, when i try update the price of the rate plan, the validation fails with  'The rate plan slug must be unique.'

```
if($model->validation() && $model->update())
{
    return $this->rest->ok($model->toArray());
}
```

I thought that this bug was fixed on 2.0.8.

**Comment by Jurigag** [comment]
Why you even doing this:

`$model->validation()` ? $model->update() is already doing `validation()`

**Comment by hisorange** [comment]
When you call validation the model does not know for what kind of operations so ofc it will fail.

**Comment by Jurigag** [comment]
Yea right, i missed it, it's obvious.

**Comment by deulos** [comment]
I confirm this is still happenning in Phalcon 2.0.13

Here is my table:
```
create table `users` (
	`id` int unsigned not null auto_increment,
	`created_at` datetime not null,
	`updated_at` datetime default null,
	`deleted_at` datetime default null,
	`alias` varchar(255) not null,
	`email` varchar(255) not null,
	`password` varchar(255) not null,
	`role` varchar(20) not null,
	primary key (`id`),
	unique key (`alias`),
	unique key (`email`)
) engine=innodb default charset=utf8 collate=utf8_unicode_ci;
```

Here the controller action:
```
public function update($id) {
	$user = Users::findFirst($id);
	
	if ($user == false)
		return $this->returnError(404, 'Not Found', 'User not found');

	$data = (array) $this->request->getJsonRawBody();

	if ($user->update($data) == false)
		return $this->returnError(400, 'Bad Request', $user->getResponseMessages());
}
```

And my model:
```
class Users extends BaseModel {

	public function initialize() {
		$this->addBehavior(new SoftDelete([
			'field' => 'deleted_at',
			'value' => date('Y-m-d H:i:s')
		]));
	}

	public function validation() {
		$this->validate(new PresenceOf([
			'field' => 'alias',
			'message' => 'Alias field is required'
		]));

		$this->validate(new Uniqueness([
			'field'   => 'alias',
			'message' => 'A user with that alias is already in use'
		]));

		$this->validate(new Email([
			'field' => 'email',
			'message' => 'Email seems to be invalid'
		]));

		$this->validate(new Uniqueness([
			'field'   => 'email',
			'message' => 'A user with the same email address is already registered, forgot the password?'
		]));

		$this->validate(new PresenceOf([
			'field' => 'password',
			'message' => 'Password field is required'
		]));

		return $this->validationHasFailed() == false;
	}


	public function beforeValidationOnCreate() {
		$this->created_at = date('Y-m-d H:i:s');
		$this->password = $this->di->get('security')->hash($this->password);

		$session = $this->di->get('session');

		if (isset($this->role) == false || $session->has('user') == false || $session->get('user')->role != 'admin')
			$this->role = 'user';
	}


	public function beforeValidationOnUpdate() {
		$this->updated_at = date('Y-m-d H:i:s');
	}
}
```

Running the update the response is:
> A user with the same email address is already registered, forgot the password?

What results weird to me is that the alias field is also unique but has no problem.

**Comment by Jurigag** [comment]
I don't know, maybe there is just such an email ?

**Comment by deulos** [comment]
This is the request I use to test it:
```
curl -i -X PUT -d '{"alias": "xxxxxx", "email": "xxxxxxxx@xxxxxxxx.com", "password": "xxxxxxxxxxxx"}' 'http://www.domain.com/users/2'
```
So the model is receiving the 3 fields but none of them really differs from database values.

By the way thanks for the tip of getJsonRawBody parameter!

**Comment by Jurigag** [comment]
And what exactly $data returns ?

**Comment by deulos** [comment]
Fool me, I was updating the user 2 instead of the 1, so the the complain about the email was right because was being used by user 1.

Just fixed it and works great now, so the bug is solved in 2.0.13

I'm sorry about the mistake, I should start using other testing data more identifiable ;P

**Comment by Jurigag** [comment]
This can be closed.
